src/dotnet/Chat.UI.Blazor/Services/ChatUI.cs

```diff
@@ -212,7 +214,8 @@ public virtual async Task<Trimmed<int>> GetUnreadCount(ChatId chatId, Cancellati
     [ComputeMethod]
     public virtual async Task<bool> IsEmpty(ChatId chatId, CancellationToken cancellationToken)
     {
-        var idRange = await Chats.GetIdRange(Session, chatId, ChatEntryKind.Text, cancellationToken).ConfigureAwait(false);
+        using var idRangeLease = await LeaseChatIdRangeState(chatId, cancellationToken).ConfigureAwait(false);
```

Comment:
So this is the only usage - can you clarify why it's necessary here?

Reply:
I also use it at ChatView to avoid dependency on ChatIdRange when we are rendering messages far from the end